app/controllers/posts_controller.rb

```diff
@@ -1,10 +1,55 @@
 class PostsController < ApplicationController
   def index
+    # This action retrieves all posts belonging to a specific user (specified by params[:user_id])
+    # and assigns them to the @posts instance variable.
+    # It's commonly used for displaying a list of posts for a particular user on the index page.
     @posts = Post.where(author_id: params[:user_id])
     @users = User.where(id: params[:user_id])
   end
 
   def show
+    # This action finds a single post based on the id parameter in the request
+    # and assigns it to the @post instance variable.
+    # It's used for displaying the details of a specific post on its own page.
     @post = Post.find(params[:id])
   end
+
+  def new
+    # This action is responsible for preparing the form to create a new post.
+    # It finds the user (specified by params[:user_id]) and assigns it to the @user instance variable.
+    # It initializes a new empty post object and assigns it to the @post instance variable.
+    @user = User.find(params[:user_id])
+    @post = Post.new
+  end
+
+  def create
+    # This action is triggered when the form for creating a new post is submitted.
+    # It sets the current user (assuming the set_current_user method is defined elsewhere).
+    # It builds a new post associated with the current user using the parameters
+    # from the form (permitted by post_params).
+    # It sets the initial values for comments_counter and likes_counter
+    # (assuming these are counters for comments and likes on the post).
+    # If the post is successfully saved, it redirects to the show page for the newly created post with a success notice.
+    # If the save fails, it renders the new template again, showing validation errors.
+
+    set_current_user
+    @post = @current_user.posts.build(post_params)
+    @post.comments_counter = 0
+    @post.likes_counter = 0
```

Comment:
- [ ] Kindly note that the class `PostController` is a child of the class `ApplicationController` and that mean it inherited all methods that is in the application_controller, Therefore we don't need to set the current user it have be already set in the `ApplicationController`. So I suggest that use remove the `set_current_user` and make use of the `current_user` method that you created in the `ApplicationController`

  Example:
```rb
@post = current_user.posts.build(post_params)
```

Reply:
Applied the suggested changes. Thanks ;)